2996_reflection/reflection.md

```diff
@@ -996,6 +997,86 @@ Thus, the only instantiations of `TU_Ticket::Helper` occur because of the call t
 
 [On Compiler Explorer](https://godbolt.org/z/1vEjW4sTr).
 
+## Emulating typeful reflection
+Although we believe a single opaque `std::meta::info` type to be the best and most scalable foundation for reflection, we acknowledge the desire expressed by SG7 for future support for "typeful reflection". The following demonstrates one possible means of assembling a typeful reflection library, in which different classes of reflections are represented by distinct types, on top of the facilities proposed here.
+
+::: bq
+```cpp
+// Represents a 'std::meta::info' constrained by a predicate.
+template <std::meta::info Pred>
+  requires (type_of(reflect_invoke(Pred, {reflect_value(^int)})) == ^bool)
+struct metatype {
+  // The underlying 'std::meta::info' value.
+  const std::meta::info value;
+
+  // Construction is ill-formed unless predicate is satisfied.
+  consteval metatype(std::meta::info r) : value(r) {
+    if (![:Pred:](r))
+      throw "Reflection is not a member of this metatype";
+  }
+
+  // Cast to 'std::meta::info' allows values of this type to be spliced.
+  consteval operator const std::meta::info&() const { return value; }
+
+  static consteval bool check(std::meta::info r) { return [:Pred:](r); }
+
+  static consteval metatype<Pred> make(std::meta::info r) { return {r}; }
+};
+
+// Type representing a "failure to match" any known metatypes.
+struct unmatched { static consteval unmatched make(...) { return {}; } };
+
+// Returns the given reflection "enriched" with a more descriptive type.
+template <typename... Choices>
+constexpr auto enrich = [](std::meta::info r) consteval {
+  std::array checks = {^Choices::check...};
+  std::array ctors = {^Choices::make...};
```

Comment:
Is there a way that we can just use the constructors? Since that's really all the `make` functions are here for.

Reply:
Done.